File: OOPROG/Sortation_Algorithms/insertionSort.c
#include <stdio.h>

void printArray(int arr[], int size);
void insertionSort(int arr[], int size);

int main(int argc, char const *argv[]){
    int size;
    int nums[] = {1, 8, 2, 6, 4, 0, 9};

    size = sizeof(nums) / sizeof(nums[0]);

    printf("Unsorted:\n");
    printArray(nums, size);

    insertionSort(nums, size);

    printf("Sorted:\n");
    printArray(nums, size);

    return 0;
}

void printArray(int arr[], int size){
    for (int i = 0; i < size; ++i) printf("%d ", arr[i]);
    printf("\n");
}

void insertionSort(int arr[], int size){
    int val, j;

    for (int i = 1; i < size; ++i) {
        val = arr[i];
        j = i - 1;

        while (j >= 0 && arr[j] > val) {
            arr[j + 1] = arr[j];
            --j;
        }
        arr[j + 1] = val;
    }
}

File: OOPROG/Sortation_Algorithms/InsertionSort.java
public class InsertionSort {
    public static void main(String[] args) {
        int nums[] = {1, 8, 2, 6, 4, 0, 9};

        System.out.println("Unsorted");
        printArray(nums);

        insertionSort(nums);

        System.out.println("Sorted:");
        printArray(nums);
    }
    static void printArray(int[] num){
        for (int i : num) {
            System.out.print(i + " ");
        }
        System.out.println();
    }
    static void insertionSort(int[] nums){
        int val, j, size;

        size = nums.length;

        for (int i = 1; i < size; i++){
            val = nums[i];

            for (j = i - 1; j >= 0 && nums[j] > val; j--)
                nums[j + 1] = nums[j];

            nums[j + 1] = val;
        }
    }
}